### PR TITLE
(PDB-3058) Strip null bytes in terminus (on the stable branch)

### PR DESCRIPTION
**Pull request by mullr**

### CODE REVIEW

File: test/puppetlabs/puppetdb/integration/inventory/basic_fact_retrieval.clj [hunk 1]
```diff
@@ -39,5 +39,15 @@
                (-> (int/pql-query pdb "facts { name = 'my_structured_fact' }")
                    first
                    (get :value)
-                   json/parse-string)))))))
+                   json/parse-string))))
 
+      (testing "Create an custom fact with a null byte"
```
Comment (rbrw) [inline test/puppetlabs/puppetdb/integration/inventory/basic_fact_retrieval.clj]: "a custom"?

File: test/puppetlabs/puppetdb/integration/inventory/basic_fact_retrieval.clj [hunk 2]
```diff
@@ -39,5 +39,15 @@
...
+        (int/run-puppet ps pdb "notify { 'irrelevant manifest': }"
+                        {:certname "null-fact-agent"
+                         :env {"FACTER_nullfact"
+                               "{\"nullfact\": \"foo\\u0000bar\"}" }}))
+
+      (testing "Ensure that the null fact is passed through properly"
+        (is (= 1
+               (-> (int/pql-query pdb "facts { name = 'nullfact' }")
+                   count)))))))
```
Comment (rbrw) [inline test/puppetlabs/puppetdb/integration/inventory/basic_fact_retrieval.clj]: Do we care about checking how the value ends up?

File: puppet/lib/puppet/util/puppetdb/char_encoding.rb [hunk 3]
```diff
@@ -123,20 +124,24 @@ def self.coerce_to_utf8(str, error_context_str=nil)
     str_copy.force_encoding("UTF-8")
     if str_copy.valid_encoding?
       str_copy.encode!("UTF-8")
+      str_copy.gsub!("\\u0000", DEFAULT_INVALID_CHAR)
```
Comment (rbrw) [inline puppet/lib/puppet/util/puppetdb/char_encoding.rb]: Should this be DEFAULT_INVALID_CHAR_ESCAPED?

**Comment by mullr** [comment]
Closing this; we're going to try a different approach.

**Pull request closed without merging**